scripts/rundocker.sh

```diff
@@ -37,6 +30,36 @@ RUN_OPTS=(
   --env ARCHES
 )
 
+# If we're running in podman, assume the user namespace is setup so that
+# root inside the container is the same as the outside user. Otherwise,
+# get the UID and GID to run as.
+if docker_is_podman; then
+  BUILD_UID=0
+  BUILD_GID=0
+else
+  BUILD_UID=$(id -u)
+  BUILD_GID=$(id -g)
+fi
+
+# If the container user is root, mount the cache at /root. Otherwise,
+# set HOME since there's likely no account with that UID in the image.
+# The user's home directory is where all the intermediate build outputs
+# (e.g., ~/.local/share/python-for-android and ~/.gradle) are stored.
+if [ "$BUILD_UID" -eq 0 ]; then
+  BUILD_CACHE_PATH=/root
+else
+  BUILD_CACHE_PATH=/cache
```

Comment:
Testing here. The changes in the `--user` parameter works for me. But the change in `BUILD_CACHE_PATH` is making the caching worse. I ran the build a second time and is taking ages. I will keep testing.